src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSegmentEntry.cs

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * <summary>
+ * Displays the high-level information for a Segment, and enables the user to play it
+ * </summary>
+ */
+public class RGSegmentEntry : MonoBehaviour
+{
+    public string segmentName;
+
+    public string description;
+    
+    // will be a file path when in-editor, or a resources path when in a packaged build
+    public string path;
+
+    public BotSequenceEntryType type;
+    
+    /**
+     * UI component fields
+     */
+    [SerializeField]
+    public Button playButton;
+    
+    [SerializeField]
+    public TMP_Text nameComponent;
+
+    [SerializeField]
+    public TMP_Text descriptionComponent;
+    
+    [SerializeField]
+    public GameObject segmentListIndicatorComponent;
+
+    /**
+     * <summary>
+     * Ensures all needed prefabs and fields are properly set. Shows an icon if this segment is of the SegmentList type
+     * </summary>
+     */
+    public void Start()
+    {
+        if (string.IsNullOrEmpty(path))
+        {
+            Debug.LogError($"RGSegmentEntry is missing its path");
+        }
+        
+        if (nameComponent != null)
+        {
+            nameComponent.text = segmentName;
+        }
+        else
+        {
+            Debug.LogError($"RGSegmentEntry is missing its nameComponent");            
+        }
+
+        if (descriptionComponent != null)
+        {
+            descriptionComponent.text = description;
+        }
+        else
+        {
+            Debug.LogError($"RGSegmentEntry is missing its descriptionComponent");
+        }
+
+        if (playButton != null)
+        {
+            playButton.onClick.AddListener(OnPlay);
+        }
+        else
+        {
+            Debug.LogError($"RGSegmentEntry is missing its playButton");
+        }
+
+        // create a tooltip containing the segment's description, but only if the description is populated
+        var tooltip = GetComponentInChildren<RGTooltip>();
+        if (tooltip != null && !string.IsNullOrEmpty(description))
+        {
+            tooltip.content = description;
+        }
+        else
+        {
+            tooltip.SetEnabled(false);
+        }
+
+        // show an icon indicating that this segment is a SegmentList (if needed)
+        if (type == BotSequenceEntryType.SegmentList)
+        {
+            if (segmentListIndicatorComponent != null)
+            {
+                segmentListIndicatorComponent.SetActive(true);
+            }
+            else
+            {
+                Debug.LogError($"RGSegmentEntry is missing its segmentListIndicatorComponent");
+            }
+        }
+    }
+
+    /**
+     * <summary>
+     * When the play button is clicked, start running the Segment and close the RGOverlay
+     * </summary>
+     */
+    private void OnPlay()
+    {
+        var botManager = RGBotManager.GetInstance();
+        if (botManager == null)
+        {
+            Debug.LogError("RGSegmentEntry cannot find the RGBotManager in its OnPlay function");

```

Comment:
In no (non-unit test) situation should this ever happen :/ 

I would rather this throw the exception for the null botManager later in the code so this doesn't pass our dev testing with us possibly missing one of these error messages.